### PR TITLE
Facial Emotion Detection

### PR DESCRIPTION
**Pull request by siddhant4ds**
# Pull Request for DL-Simplified 💡

## Issue Title :  Facial Emotion Detection

- **Info about the related issue (Aim of the project)** :  Training a deep learning model to classify 4 facial emotions from 48x48 grayscale images.
- **Name:**  Siddhant Tiwari
- **GitHub ID:**  https://github.com/siddhant4ds
- **Email ID:**  siddhant.tiwari.ds@gmail.com
- **Idenitfy yourself: (Mention in which program you are contributing in. Eg. For a JWOC 2022 participant it's, `JWOC Participant`)**  SWOC-S4 (2024)

Closes: #232 

### Describe the add-ons or changes you've made 📃

Give a clear description of what have you added or modifications made

## Type of change ☑️

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ⚙️

Changes made and tested on Jupyter (Google Colab and Kaggle) notebooks. [Requires GPU] 

## Checklist: ☑️

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by abhisheks008 — CHANGES_REQUESTED** [review]
The project folder name should be same as the issue name with no hyphen and underscore.

File: Facial Emotion Detection/Model/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Project Summary
+
+**PROJECT TITLE:** Facial Emotion Detection
+
+**GOAL:** Multiclass classification of facial emotions from grayscale images.
+
+**DATASET:** [FER-DS - Kaggle](https://www.kaggle.com/datasets/mhantor/facial-expression)
+
+**DESCRIPTION:**  
+We have 19950 images covering 4 facial emotions (angry, happy, neutral, surprised).
```
Comment (abhisheks008) [inline Facial Emotion Detection/Model/README.md:10]: Use bullet points/numbering.

File: Facial Emotion Detection/Model/README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+| CNN + Simple data-augment | 75.45 | 74.99 |
+| CNN + Complex data-augment | 67.82 | 67.92 |
+| CNN + Keras-CV RandAugment | 60.47 | 60.15 |
+| MobileNetV2 | 80.01 | 79.30 |
+| MobileNetV2 + Simple data-augment | 82.41 | 81.35 |
+| EfficientNetV2S | 83.24 | 82.31 |
+| EfficientNetV2S + Simple data-augment | **84.24** | **82.86** |
+
+**CONCLUSION:**  
+We used low-resolution, grayscale images to train a deep learning model to detect four facial emotions.
```
Comment (abhisheks008) [inline Facial Emotion Detection/Model/README.md:61]: Use bullet points/numbering.

**Review by abhisheks008 — CHANGES_REQUESTED** [review]
Your overall project is looking good to me. Need some minor changes in the README file.

**Comment by siddhant4ds** [comment]
@abhisheks008 I have updated the formatting in the project summary. Please let me know if any other changes are needed.

**Review by abhisheks008 — APPROVED** [review]
Approved @siddhant4ds under SWOC S4.

**Pull request merged**